File: shared/contracts/ts/v1/block.ts
export interface BlockNode {
  block_uid: string;
  note_id: string;
  parent_block_uid: string | null;
  sibling_order: number;
  block_index: number;
  content_text: string;
  rich_content: Record<string, unknown>;
}

export interface ListBlocksResponse {
  items: BlockNode[];
}

export interface BlockSearchItem {
  block_uid: string;
  note_id: string;
  note_title: string;
  content_text: string;
}

export interface BlockSearchResponse {
  items: BlockSearchItem[];
}

export interface BlockBacklinkItem {
  source_block_uid: string;
  source_note_id: string;
  source_note_title: string;
  snippet: string;
  updated_at: string;
}

export interface BlockBacklinksResponse {
  block_uid: string;
  items: BlockBacklinkItem[];
}

File: shared/contracts/python/v1/block.py
from __future__ import annotations

from pydantic import BaseModel, Field


class BlockNode(BaseModel):
    block_uid: str = Field(min_length=1)
    note_id: str = Field(min_length=1)
    parent_block_uid: str | None = None
    sibling_order: int = Field(ge=0)
    block_index: int = Field(ge=0)
    content_text: str
    rich_content: dict[str, object]


class ListBlocksResponse(BaseModel):
    items: list[BlockNode]


class BlockSearchItem(BaseModel):
    block_uid: str = Field(min_length=1)
    note_id: str = Field(min_length=1)
    note_title: str = Field(min_length=1)
    content_text: str


class BlockSearchResponse(BaseModel):
    items: list[BlockSearchItem]


class BlockBacklinkItem(BaseModel):
    source_block_uid: str = Field(min_length=1)
    source_note_id: str = Field(min_length=1)
    source_note_title: str = Field(min_length=1)
    snippet: str
    updated_at: str = Field(min_length=1)


class BlockBacklinksResponse(BaseModel):
    block_uid: str = Field(min_length=1)
    items: list[BlockBacklinkItem]

